Remove now unused cmake

K20shores opened the pull request:
With #2847 and the work done by @ZedThree, I believe we can now

- close #1444
- close 2869

When I make and install netcdf on a mac and linux, I don't see the `hdf5::hdf5-shared` libraries included in `<install-prefix>/lib/pkgconfig/netcdf.pc` and `<install-prefix>/lib/cmake/netCDF/netCDFConfig.cmake` only includes a call to `find_dependency(HDF5)` (as configured). 

My bet is that #2847 with its updates to finding and using hdf5, and updated hdf5 and cmake versions fixed this.

Related is a [spack PR](https://github.com/spack/spack/pull/42878) that should now be redundant, along with the filter code that I removed

Chrismarsh commented:
Great to see!
Further work was done on this in this [Spack PR](https://github.com/spack/spack/pull/43087).
Will these changes be back-ported or will they be only for upcoming netcdf releases?

WardF commented:
This looks good, but isn't quite there. We need to sort out the subsequent lack of spacing in `libnetcdf.settings`, and the output of `nc-config --libs --static`, e.g.:

````
Extra libraries:	-lHDF5::HDF5;-lhdf5::hdf5_hl;-lm;-lz;-lblosc;-lzstd;-lbz2;-lsz;-lcurl;-lxml2

````

and 

````
wfisher@shetland:~/Desktop/tmp/tmp/netcdf-c/build$ ./nc-config --static --libs
-L/usr/local/lib -lnetcdf -lHDF5::HDF5;-lhdf5::hdf5_hl;-lm;-lz;-lblosc;-lzstd;-lbz2;-lsz;-lcurl;-lxml2
````

WardF reviewed (changes requested):
This looks good but leaves undesired formatting in `nc-config --libs --static`.  We can't assume the end user who is invoking `nc-config` will be working within a `cmake`-focused ecosystem.

WardF commented:
> Great to see! Further work was done on this in this [Spack PR](https://github.com/spack/spack/pull/43087). Will these changes be back-ported or will they be only for upcoming netcdf releases?

They'll only be in upcoming releases; I'd love to backport but we are resource constrained, unfortunately. Hopefully we'll have it out shortly! That's the goal at least.

K20shores commented:
> This looks good, but isn't quite there. We need to sort out the subsequent lack of spacing in `libnetcdf.settings`, and the output of `nc-config --libs --static`, e.g.:
> 
> ```
> Extra libraries:	-lHDF5::HDF5;-lhdf5::hdf5_hl;-lm;-lz;-lblosc;-lzstd;-lbz2;-lsz;-lcurl;-lxml2
> ```
> 
> and
> 
> ```
> wfisher@shetland:~/Desktop/tmp/tmp/netcdf-c/build$ ./nc-config --static --libs
> -L/usr/local/lib -lnetcdf -lHDF5::HDF5;-lhdf5::hdf5_hl;-lm;-lz;-lblosc;-lzstd;-lbz2;-lsz;-lcurl;-lxml2
> ```

@WardF fixed. I replaced a thing that replaced colons with spaces that I had removed, assuming it wasn't useful
